packages/cli/src/commands/hydrogen/deploy.ts

```diff
@@ -673,3 +679,42 @@ Continue?`.value,
 
   return deployPromise;
 }
+
+/**
+ * Gets the current @shopify/hydrogen version from the app's package.json
+ */
+export async function getHydrogenVersion({appPath}: {appPath: string}) {
+  const {root} = getProjectPaths(appPath);
+
+  let packageJson: PackageJson | undefined;
+  const nodeModulesHydrogenPath = joinPath(
+    root,
+    'node_modules',
+    '@shopify',
+    'hydrogen',
+    'package.json',
```

Comment:
Will do a more in depth review, but I'm not sure this will work in some package managers like yarn >2 and pnpm. Could you try testing?

Reply:
Yeah, it looks like the node_modules look a lot different in yarn@latest. pnpm works, and I think my approach is here is trying to do the best we can and fall back to different options. But open to more ideas.

For example below , if they have `^` or `>` in package.json for @shopify/hydrogen it can be slightly less accurate too, but we're trying to get the best data we can basically. 

We can also rely on parsing the working bundle and crawling it as a fallback too if we don't like any of these. This is being used to backfill data too.

Reply:
@benwolfram I think we can use `localImport` that's being used within the Hydrogen CLI repo to get the Hydrogen lib version accurate to the node package itself.

1. Export the `LIB_VERSION` in hydrogen package - This lib version will be updated on every release including if someone is using a next tag (which won't have a proper version tag and we can ignore showing banner in admin)
2. Then in CLI, we can local import hydrogen to get the lib version like this

```tsx
export async function getHydrogenVersion({appPath}: {appPath: string}) {
  const {root} = getProjectPaths(appPath);

  type HydrogenType = typeof import('@shopify/hydrogen');
  const {LIB_VERSION} = await importLocal<HydrogenType>(
    '@shopify/hydrogen',
    root,
  ).catch(() => {
    return '1.0.0';
  });

  return LIB_VERSION;
}
```

Reply:
>I am already trying to do that in the PR, but are you saying this solution would solve for Yarn>2 as well?

The code I posted relies on Node's resolution logic, which should be safer than finding the file manually. I'd expect this to work across package managers, just like any `import` or `require` in Node work with all of them because it uses the same algorithm.

Also, no need to use `async` / `await` with this code 👍